### PR TITLE
add Jupyter to ml dev role

### PR DESCRIPTION
**Pull request by yousefiparsa**
Added Jupyter competency and use it in Engineering ML Developer Role.

### CODE REVIEW

File: competencies/jupyter.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+Project Jupyter is a nonprofit organization created to "develop open-source software, open-standards, and services for interactive computing across dozens of programming languages". Spun-off from IPython in 2014 by Fernando Pérez, Project Jupyter supports execution environments in several dozen languages. Project Jupyter's name is a reference to the three core programming languages supported by Jupyter, which are Julia, Python and R, and also a homage to Galileo's notebooks recording the discovery of the moons of Jupiter. Project Jupyter has developed and supported the interactive computing products Jupyter Notebook, JupyterHub, and JupyterLab, the next-generation version of Jupyter Notebook.
+
+Useful links:
+
+- Wikipedia: https://en.wikipedia.org/wiki/Project_Jupyter
+
+## How do you prove it?
+
+- You know what is JupyterLab, JupyterHub, Jupyter Console, and a Jupyter Notebook, and what the main differences they have.
```
Comment (codeallthethingz) [inline competencies/jupyter.md:11]:
```suggestion
- You know what a JupyterLab, JupyterHub, Jupyter Console, Jupyter Notebook is and the main differences between them.
```

File: competencies/jupyter.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+- You know what is Google Colab, and what differences it has with Jupyter environment.
```
Comment (codeallthethingz) [inline competencies/jupyter.md:13]:
```suggestion
- You know what a Google Colab is and the differences between a Jupyter environment.
```

File: competencies/jupyter.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+- You know what are the differences of a Jupyter notebook with a programming script.
```
Comment (codeallthethingz) [inline competencies/jupyter.md:15]:
```suggestion
- You know what the differences between a Jupyter notebook and a programming script are.
```

File: competencies/jupyter.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+- You know how to install Jupyter products using Anaconda or PIP package manager.
+
+- You know how to run Jupyter from terminal with all different modes using different flags.
```
Comment (codeallthethingz) [inline competencies/jupyter.md:19]:
```suggestion
- You know how to run Jupyter from the terminal with all different modes using different flags.
```

File: competencies/jupyter.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+- You know Jupyter shortcuts.
```
Comment (codeallthethingz) [inline competencies/jupyter.md:21]: Lets give it a value?  5?

File: competencies/jupyter.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+- You know what is Command mode and Edit mode, and how to use each one.
```
Comment (codeallthethingz) [inline competencies/jupyter.md:23]:
```suggestion
- You know what Command mode and Edit mode are, and how to use each one.
```

File: competencies/jupyter.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+- You know how to evaluate a cell without mouse.
```
Comment (codeallthethingz) [inline competencies/jupyter.md:25]:
```suggestion
- You know how to evaluate a cell without a mouse.
```

File: competencies/jupyter.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+- You know what are Jupyter Kernels, what they do, and how to switch between them.
```
Comment (codeallthethingz) [inline competencies/jupyter.md:27]:
```suggestion
- You know what Jupyter Kernels are, what they do, and how to switch between them.
```

File: competencies/jupyter.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+- You know how to restart, stop, and shut down a kernel.
+
+- You know how to use terminal in Jupyter and what differences it has with your OS terminal.
+
+- You know how to set up extensions and use them in your Jupyter environment.
+
+- You know how to activate X11 on Jupyter environment.
```
Comment (codeallthethingz) [inline competencies/jupyter.md:35]:
```suggestion
- You know how to activate X11 on your Jupyter environment.
```

File: competencies/jupyter.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+- You know how to share your code in Jupyter with other colleagues.
+
+- You know how to use document mode and presentation mode for presenting your code in Jupyter.
+
+- You know how to switch between your launcher panel to your notebooks without using mouse.
```
Comment (codeallthethingz) [inline competencies/jupyter.md:41]:
```suggestion
- You know how to switch between your launcher panel to your notebooks without using a mouse.
```

**Comment by codeallthethingz** [comment]
Good work! just some minor grammatical changes.

File: competencies/jupyter.md (hunk 2 above)
Comment (yousefiparsa) [reply competencies/jupyter.md:13]: Google Colab is a specific name of an environment launched by Google. I don't think it gets an "a".

File: competencies/jupyter.md [hunk 11]
```diff
@@ -0,0 +1,61 @@
+# Competency - Jupyter
+
+Project Jupyter is a nonprofit organization created to "develop open-source software, open-standards, and services for interactive computing across dozens of programming languages". Spun-off from IPython in 2014 by Fernando Pérez, Project Jupyter supports execution environments in several dozen languages. Project Jupyter's name is a reference to the three core programming languages supported by Jupyter, which are Julia, Python and R, and also a homage to Galileo's notebooks recording the discovery of the moons of Jupiter. Project Jupyter has developed and supported the interactive computing products Jupyter Notebook, JupyterHub, and JupyterLab, the next-generation version of Jupyter Notebook.
```
Comment (grrilla) [inline competencies/jupyter.md:3]: As a person who is unfamiliar with Jupyter, I really enjoyed this read. Some fascinating trivia! In my opinion, I think what may be missing from this opening is why Jupyter is valuable for this role at all. Why is Jupyter so widely used in the ML world if it isn't ML-specific? What are the benefits to adopting its use as a member of this team? How should it be used? These are just examples, but hopefully they help to illustrate the sorts of questions we generally want to answer surrounding why this competency might be considered important.

Comment (grrilla) [reply competencies/jupyter.md:3]: Oh, and it wouldn't need to be something you'd go at length about. Just a couple sentences for context.

Comment (yousefiparsa) [reply competencies/jupyter.md:3]: A great point, thanks. I would add that.

**Pull request merged**